python/pycylon/data/table.pyx

```diff
@@ -198,38 +206,107 @@ cdef class Table:
         s = Status(status.get_code(), b"", -1)
         return s
 
-    def join(self, ctx: CylonContext, table: Table, join_type: str, algorithm: str, left_col: int,
-             right_col: int) -> Table:
+    def _is_pycylon_table(self, obj):
+        return isinstance(obj, Table)
+
+    def _resolve_column_indices_from_column_names(self, column_names: List[
+        str], op_column_names: List[str]) -> List[int]:
+        resolve_col_ids = []
+        for op_col_id, op_column_name in enumerate(op_column_names):
+            for col_id, column_name in enumerate(column_names):
+                if op_column_name == column_name:
+                    resolve_col_ids.append(col_id)
+        return resolve_col_ids
```

Comment:
what should we do if the provided column name strings are not found?

Reply:
Let's throw an exception

Reply:
@supunkamburugamuve @nirandaperera there is already a check within join and distjoin to verify if there are any elements in the returned list. If no items are there an exception is thrown. 

But I am going to make it neater and pushing it to column retrieval and add a function call.

Reply:
@supunkamburugamuve @nirandaperera 
Updated.